fetch.js

```diff
@@ -217,6 +217,11 @@
 
   function Request(url, options) {
     options = options || {}
+
+    if (url && url.match(/^([\w-]+:)?\/\/[^\/]+@/)) {
```

Comment:
> I'm not super familiar with the behavior of Fetch. If this isn't security sensitive, then a regex might be good enough.

Thanks for jumping in. Yes, this isn't security-sensitive. If you try to use a URL that contains username/password with XHR, the username/password gets ignored because XHR wants you to pass them as arguments to `xhr.open()`.

The Fetch spec contemplated supporting username/password in URL strings but dropped that idea because of the complexities around cross-origin redirects. So, this PR adds a helpful TypeError exception so you know what's goin on. If the URL string is super-weird and the regex fails, you simply won't get a helpful exception (and no auth credentials will be sent).